src/Vision/ImageAnnotatorTrait.php

```diff
@@ -0,0 +1,167 @@
+
+        $maxResults = $optionalArgs['maxResults'];
+        $imageContext = $optionalArgs['imageContext'];
+        $requestClass = $optionalArgs['requestClass'];
+        $featureClass = $optionalArgs['featureClass'];
+
+        $annotateImageRequest = new $requestClass();
+        $annotateImageRequest->setImage($this->resolveImage($image));
+
+        foreach ($features as $featureType) {
+            $feature = new $featureClass();
+            $feature->setType($featureType);
+            if (isset($maxResults[$featureType])) {
+                $feature->setMaxResults($maxResults[$featureType]);
+            }
+            $annotateImageRequest->addFeatures($feature);
+        }
+        if (isset($imageContext)) {
+            $annotateImageRequest->setImageContext($imageContext);
+        }
+
+        $optionalArgs = array_diff_key($optionalArgs, $defaultArgs);
+        $batchResponse = $this->client->batchAnnotateImages([$annotateImageRequest], $optionalArgs);
+        return $batchResponse->getResponses(0);
+    }
+
+    /**
+     * @param resource|string|Image $image
+     * @param \Google\Cloud\Vision\V1\Feature\Type|int $featureType
+     * @param string $getFeatureMethod
+     * @param array $optionalArgs {
+     *
+     *     @type integer $maxResults A maximum number of features to return
+     *     @type \Google\Cloud\Vision\V1\ImageContext $imageContext
+     *     @type \Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for this call. If present, then
+     *          $timeoutMillis is ignored.
+     *     @type int $timeoutMillis
+     *          Timeout to use for this call. Only used if $retrySettings
+     *          is not set.
+     * }
+     * @return FaceAnnotation[] Array of face annotations
+     */
+    protected function annotateFeatureTypeImpl($image, $featureType, $getFeatureMethod, $optionalArgs = [])
+    {
+        $maxResults = [];
+        if (isset($optionalArgs['maxResults'])) {
+            $maxResults[$featureType] = $optionalArgs['maxResults'];
+        }
+        $optionalArgs['maxResults'] = $maxResults;
+        $response = $this->annotateImage($image, [$featureType], $optionalArgs);
+        return $response->$getFeatureMethod();
+    }
+
+    /**
+     * @param FaceAnnotation $faceAnnotation
+     * @param FaceAnnotation\Landmark\Type|int $landmarkType
+     * @return Position|null Landmark position
+     */
+    protected function getFaceLandmarkPositionImpl($faceAnnotation, $landmarkType)
+    {
+        foreach ($faceAnnotation->getLandmarksList() as $landmark) {
+            if ($landmark->getType() === $landmarkType) {
+                return $landmark->getPosition();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param resource|string|Image $imageInput
+     * @param string $imageObjClass
+     * @param string $imageSourceClass
+     * @return Image
+     */
+    protected function resolveImage(
```

Comment:
Can you explain in plain English what this function is doing and why it exists? Is it just making it so that file-like objects work (line 157-158)? Is there a lighter approach?